packages/desmos-std/src/msg_router.rs

```diff
@@ -0,0 +1,177 @@
+use crate::{
+    profiles::{
+        msg_router::ProfilesMsgRouter,
+        msg_routes::ProfilesMsgs,
+        msg_routes::ProfilesMsgs::{
+            AcceptDtagTransferRequest, BlockUser, CancelDtagTransferRequest, CreateRelationship,
+            DeleteProfile, DeleteRelationship, RefuseDtagTransferRequest, RequestDtagTransfer,
+            SaveProfile, UnblockUser,
+        },
+    },
+    types::DesmosRoute,
+};
+use cosmwasm_std::{Addr, CosmosMsg, CustomMsg};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct DesmosMsgRouter {
+    pub route: DesmosRoute,
+    pub msg_data: DesmosMsg,
+}
+
+impl Into<CosmosMsg<DesmosMsgRouter>> for DesmosMsgRouter {
+    fn into(self) -> CosmosMsg<DesmosMsgRouter> {
+        CosmosMsg::Custom(self)
+    }
+}
+
+impl CustomMsg for DesmosMsgRouter {}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[non_exhaustive] // missing chain-links and app-links messages (not necessary to me)
+pub enum DesmosMsg {
+    Profiles(ProfilesMsgs),
+}
+
+impl ProfilesMsgRouter<DesmosMsgRouter> for DesmosMsgRouter {
```

Comment:
To build the `MsgRouter`, we should create empty `MsgRouter` firstly then rebuild it by the functions in the trait. The process is a bit strange to me.

I propose to have a `MsgRouterBuilder` to handle the `MsgRouter` creation, which is like as follows:

```rust
pub struct DesmosMsgRouterBuilder {}
impl DesmosMsgRouterBuilder{
    pub fn new() -> Self {
        DesmosMsgRouterBuilder{}
    }
}

impl ProfilesMsgRouterBuilder<DesmosMsgRouter> for DesmosMsgRouterBuilder {
    fn save_profile(...) -> DesmosMsgRouter {
        ...
    }
    ...
}
```

What do you think? @bragaz

Reply:
@dadamu makes sense! 
I would call it `DesmosMsgBuilder`, since it builds messages and not the router itself.